DSCResources/MSFT_xDSCWebService/SecureTLSProtocols.psm1

```diff
@@ -0,0 +1,48 @@
+# This module file contains a utility to test and set TLS protocol according best practices
+#
+# Copyright (c) Microsoft Corporation, 2016
+#
+
+# ============ Best Practice Security Settings Block =========
+$insecureProtocols            = @("SSL 2.0", "SSL 3.0", "TLS 1.0", "PCT 1.0", "Multi-Protocol Unified Hello")
+$secureProtocols              = @("TLS 1.1", "TLS 1.2")
+
+# ===========================================================
+
+function Test-Protocol
+{
+    foreach ($protocol in $insecureProtocols)
+    {
+        $registryPath = "HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\$protocol\Server"
+        if (($null -ne (Get-Item -Path $registryPath -ErrorAction SilentlyContinue)) -and ($null -ne (Get-ItemProperty -Path $registryPath)) -and ((Get-ItemProperty -Path $registryPath).Enabled -ne 0))
+        {
+            return $false
+        }
+    }
+    foreach ($protocol in $secureProtocols)
+    {
+        $registryPath = "HKLM:\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\$protocol\Server"
+        if (($null -eq (Get-Item -Path $registryPath -ErrorAction SilentlyContinue -ErrorVariable ev)) -or ($null -eq (Get-ItemProperty -Path $registryPath)) -or ((Get-ItemProperty -Path $registryPath).Enabled -eq 0))
```

Comment:
>SilentlyContinue [](start = 66, length = 16)

Should we test the path first instead of ignoring errors? I am thinking that if Get-Item fails for some other reason (e.g. using runAs with an account that doesn't have access to the registry) we would not report any error.